File: k01-logica-de-programacao/cap-06/arrays/EmbaralhaArray.cs
/* VERSAO 1 */
/*
class EmbaralhaArray
{
	static void Main ()
	{
		int [] array = new int [10];

		System.Random geradorDeNumeroAleatorio = new System.Random();

		for (int i = 0; i < array.Length; i ++)
		{
		
			array[i] = i;
		}

		for (int i = 0; i < array.Length; i ++)
		{
			
			System.Console.WriteLine(array[i]);
		}
		for (int i = 0; < 10; i ++)
		{
			
			int posicao1 = geradorDeNumeroAleatorio.Next(0,11);
			int posicao2 = geradorDeNumeroAleatorio.Next(0,11);
			int auxiliar = array[posicao1];
							
			array[posicao1] = array[posicao2];
			array[posicao2] = auxiliar;
		}
		System.Console.WriteLine("--------------------");

		for (int i = 0; i < array.Length; i ++)
		{
			
			System.Console.WriteLine(array[i]);
		}
	}
}
*/

File: k02-desenvolvimento-web-com-html-css-e-javascript/cap-04/javascript/embaralha-array.js
var array = new Array(10);
for (var x = 0; x < array.length; x++) {
	array[x] = x;
}

for (var y = 0; y < array.length; y++) {
	document.writeln(array[y]);
	document.writeln('<br/>');
}

for(var z = 0; z < 15; z++) {
	var posicao1 = Math.floor(Math.random()*10);
	var posicao2 = Math.floor(Math.random()*10);
	var auxiliar = array[posicao1];
	
	array[posicao1] = array[posicao2];
	array[posicao2] = auxiliar;
}

document.writeln("------------------");
document.writeln('<br/>');

for(var w = 0; w < array.length; w++) {
	document.writeln(array[w]);
	document.writeln('<br/>');
}